json/schemas/whitelabel-schema.json

```diff
@@ -0,0 +1,119 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "title": "White Labelling",
+    "description": "White Labelling Configuration, see section 5.7 in the RP2350 datasheet for more details",
+    "type": "object",
+    "properties": {
+        "$schema": {},
+        "device": {
+            "description": "Device Properties",
+            "type": "object",
+            "properties": {
+                "vid": {
+                    "description": "Vendor ID",
+                    "type": "string",
+                    "pattern": "^0x[0-9a-fA-F]{4}$"
+                },
+                "pid": {
+                    "description": "Product ID",
+                    "type": "string",
+                    "pattern": "^0x[0-9a-fA-F]{4}$"
+                },
+                "bcd": {
+                    "description": "Device Revision",
+                    "type": "number",
+                    "minimum": 0,
+                    "maximum": 99
+                },
+                "lang_id": {
+                    "description": "Language ID",
+                    "type": "string",
+                    "pattern": "^0x[0-9a-fA-F]{4}$"
+                },
+                "manufacturer": {
+                    "description": "Manufacturer Name (can contain unicode)",
+                    "type": "string",
+                    "maxLength": 30
+                },
+                "product": {
+                    "description": "Product Name (can contain unicode)",
+                    "type": "string",
+                    "maxLength": 30
+                },
+                "serial_number": {
+                    "description": "Serial Number (can contain unicode)",
+                    "type": "string",
+                    "maxLength": 30
+                },
+                "max_power": {
+                    "description": "Max power consumption, in 2mA units",
+                    "type": ["integer", "string"],
+                    "maximum": 255,
+                    "pattern": "^0x[0-9a-fA-F]{0,2}$"
+                },
+                "attributes": {
+                    "description": "Device attributes: bit 7 must be 1, bit 6 is self-powered, bit 5 is remote wakeup, bits 0-4 must be 0",
```

Comment:
If bit 7 must be 1, does that imply that 128 is the minimum value? And if bits 0-4 must be 0, then I guess 255 isn't the maximum value?

Reply:
Fixed - the range is 128-224. Technically only the 4 values `0x80`, `0xA0`, `0xC0` and `0xE0` are allowed, so I've changed the regex to match that